app/services/decorated_session.rb

```diff
@@ -0,0 +1,18 @@
+class DecoratedSession
+  def initialize(sp:, view_context:)
+    @sp = sp
+    @view_context = view_context
+  end
+
+  def call
+    if sp.is_a? ServiceProvider
```

Comment:
This is needed because `ServiceProvider.from_issuer` returns `NullServiceProvider` if no DB matches were found. From a session decorator standpoint, we want to treat a `NullServiceProvider` the same as the absence of an SP.